build/common.project.props

```diff
@@ -14,9 +14,13 @@
     <NETFXTargetFrameworkVersion>v4.7.2</NETFXTargetFrameworkVersion>
     <NETFXTargetFramework>net472</NETFXTargetFramework>
     <NETCoreTargetFramework>netcoreapp2.1</NETCoreTargetFramework>
+    <NETCoreTargetFrameworks Condition=" '$(RequiresSigningXplatAPIs)' != 'true' ">netcoreapp2.1</NETCoreTargetFrameworks>
+    <NETCoreTargetFrameworks Condition=" '$(RequiresSigningXplatAPIs)' == 'true' ">netcoreapp2.1;netcoreapp5.0</NETCoreTargetFrameworks>
+    <NETCoreTestTargetFrameworks>netcoreapp3.1;netcoreapp5.0</NETCoreTestTargetFrameworks>
     <NetStandardVersion>netstandard2.0</NetStandardVersion>
-    <TargetFrameworksExe>$(NETFXTargetFramework);$(NETCoreTargetFramework)</TargetFrameworksExe>
-    <TargetFrameworksLibrary>$(NETFXTargetFramework);$(NetStandardVersion)</TargetFrameworksLibrary>
+    <TargetFrameworksExe>$(NETFXTargetFramework);$(NETCoreTargetFrameworks)</TargetFrameworksExe>
+    <TargetFrameworksLibrary  Condition=" '$(RequiresSigningXplatAPIs)' != 'true' ">$(NETFXTargetFramework);$(NetStandardVersion)</TargetFrameworksLibrary>
+    <TargetFrameworksLibrary  Condition=" '$(RequiresSigningXplatAPIs)' == 'true' ">$(NETFXTargetFramework);$(NetStandardVersion);netcoreapp5.0</TargetFrameworksLibrary>
```

Comment:
I think I've talked to you about when I worked on the dotnet.integration.tests multi-tfm testing: I'm not a fan of defining `RequiresSigningXplatAPIs` in the project, importing this props file, then using `TargetFrameworksLibrary` in the `csproj`'s `TargetFrameworks` element.

In my opnion, either:
* set properties in the project, import this file, and this file sets the `TargetFrameworks`

or

* import this file, which defines a bunch of different properties, and the project file uses the one it likes in its own definition of `TargetFrameworks`.

I prefer the second option, which is how `NuGet.Client` worked before the `dev-feature-xplatVerification` branch, but this current approach where some of the TFM logic is in the project file and other TFM logic is in the common project props file, it makes it harder to understand.

However, it's something we can fix after this branch is merged into `dev`.

Reply:
Thanks. Will fix in #3571